lib/dns.c

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2020 Netfoundry, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <uv_mbed/queue.h>
+
+
+#define MAX_DNS_NAME 256
+#define MAX_IP_LENGTH 16
+
+struct dns_entry {
+    char hostname[MAX_DNS_NAME];
+    char ip[MAX_IP_LENGTH];
+
```

Comment:
Should there be a flag here to indicate the record type (A vs AAAA)?

Reply:
right now MAX_IP_LENGTH is 16, no support for IPv6 dns assignments. we can add those later